### PR TITLE
Disable interactions via VisCanvas

### PR DESCRIPTION
**Pull request by loichuder**
Work towards #1020 

The last step will be to allow interactions to be changed through a prop of `...Vis` components that will be forwarded to `VisCanvas`

### CODE REVIEW

File: packages/lib/src/vis/rgb/RgbVis.tsx [hunk 1]
```diff
@@ -59,12 +58,18 @@ function RgbVis(props: Props) {
...
           flip: true,
         }}
-        interactionKeys={DEFAULT_INTERACTIONS_KEYS}
+        interactionKeys={{
+          Pan: true,
+          Zoom: true,
+          XAxisZoom: { modifierKey: 'Alt', disabled: keepRatio },
+          YAxisZoom: { modifierKey: 'Shift', disabled: keepRatio },
+          SelectToZoom: { modifierKey: 'Control' },
+        }}
```
Comment (loichuder) [inline packages/lib/src/vis/rgb/RgbVis.tsx:67]: I could not use `DEFAULT_INTERACTIONS_KEYS` here due to dynamic setting of `disabled`.

Do you think it is worth putting the creating of an object in an utility ?

File: packages/lib/src/vis/shared/VisCanvas.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ interface Props {
   visRatio?: number | undefined;
   abscissaConfig: AxisConfig;
   ordinateConfig: AxisConfig;
-  interactionKeys?: InteractionKeys;
+  interactionKeys?: InteractionParams;
```
Comment (axelboc) [inline packages/lib/src/vis/shared/VisCanvas.tsx:19]: What do you think of `interactions: Interactions` or `InteractionsConfig`?

**Review by axelboc — APPROVED** [review]

File: packages/lib/src/vis/shared/VisCanvas.tsx (hunk 2 above)
Comment (loichuder) [reply packages/lib/src/vis/shared/VisCanvas.tsx:19]: I would like to put `Interactions` but there is a `Interaction` interface for the interaction help button that is different. This being said, we can rename `Interaction` into `InteractionHelp` or something like that

File: packages/lib/src/interactions/Pan.tsx [hunk 3]
```diff
@@ -47,7 +38,7 @@ function Pan(props: Props) {
 
   const onPointerMove = useCallback(
     (evt: CanvasEvent<PointerEvent>) => {
-      if (disabled || !startOffsetPosition.current) {
+      if (!startOffsetPosition.current) {
```
[the remark this answers is not in the document]
Comment (loichuder) [reply packages/lib/src/interactions/Pan.tsx:41]: Not too sure: the movement of the camera feels odd when releasing, moving the mouse and pressing again.

[the remark this answers is not in the document]
Comment (axelboc) [reply packages/lib/src/interactions/Pan.tsx:41]: Hmm yeah, maybe more a use case for `useKeyboardEvent` and reseting the `startOffsetPosition` when the modifier key is released. User then has to release the pointer and restart a drag gesture for the pan to start again. For another PR.

File: packages/lib/src/vis/shared/VisCanvas.tsx (hunk 2 above)
Comment (loichuder) [reply packages/lib/src/vis/shared/VisCanvas.tsx:19]: Renamed `InteractionParams` to `Interactions` and `Interaction` to `InteractionInfo`

**Pull request merged**